src/cdomains/intDomain.ml

```diff
@@ -769,7 +788,9 @@ struct
   (* Default behaviour for unary operators, simply maps the function to the
    * DefExc data structure. *)
   let lift1 f x = match x with
-    | `Excluded (s,r) -> `Excluded (S.map f s, r)
+    | `Excluded (s,r) ->
+      let s' = S.map f s in
```

Comment:
Imo, this is still unsound:

- The range can also change if the set of excluded values is empty: Consider e.g. `neg`

~~~c
  int test = 0;
  int top;
  if(top) {
    test = 2;
  }
  int test2 = -test; 
  // We currently have test2 -> Excluded {} [0,8]
~~~

- I don't think changing the range so it fits whatever the excluded values map to is enough anyway: The values that are included might map to things that require a larger range than what we could exclude, right?

Reply:
Right, idk why I did it differently for `lift1` - should just be the same as for `lift2_inj`, i.e., apply f on the range.

Reply:
Done.